### DISCUSSION TITLE
Extension deactivate() function seems to not run/go through the logic in it.

### DISCUSSION THREAD

**Discussion by ccastrotrejo** [microsoft/vscode-discussions#1609] [Extension Development QnA] [answered]
Hi, 
For context we are from the Azure Logic Apps team, we own a vscode extension and we handle different types of dependencies, one of them is azure function core tools and we run a function core tools process when we activate the extension.

We implement a function inside the deactivate() main function in order to kill the azure function core tools process that we previously activated, however we have been experiencing some issues that  whenever we close vscode, the vscode project or the vscode workspace, not all the times the function inside it runs it.

This is the exact function we execute in the deactivate function() https://github.com/Azure/LogicAppsUX/blob/2937ed07a18c15c3c36670b1905359d07fdd5208/apps/vs-code-designer/src/main.ts#L106

Do you know if there is any way we can make sure 100% that this functions gets executed before we close the project, workspace or vscode?

Thanks
Carlos Castro

Adding @lambrianmsft to be aware

**Comment by gjsjohnmurray** [comment, marked as answer]
Something similar came up in a previous discussion here and linked to https://github.com/microsoft/vscode/issues/144118

**Comment by lambrianmsft** [comment]
@gjsjohnmurray, 
Going through the linked discussions, we aren't calling any functions asynchronously and still the cleanup doesn't happen all the time. 
Here is our disposal function. 
<img width="557" alt="image" src="https://github.com/user-attachments/assets/978dc9d6-f408-466c-a6b3-bc22bf9566b3">
Even when we are in debug mode, the deactivate function doesn't always call our cleanup function.

Cleanup also doesn't happen in these cases:
1) Ctrl + Shift + P and selecting the Reload Window command.
2) When a workspace is opened through the "Open Workspace From File" option 
<img width="245" alt="image" src="https://github.com/user-attachments/assets/f9c5762a-9329-4e43-a134-120973fc2e03">
